cpp/lib/Authenticator.cpp

```diff
@@ -112,9 +121,11 @@ extern "C" {
     }
     catch (...)
     {
-      SET_SNOWFLAKE_ERROR(&conn->error, SF_STATUS_ERROR_GENERAL,
-                          "authentication failed",
-                          SF_SQLSTATE_GENERAL_ERROR);
+        if (getAuthenticatorType(conn->authenticator) == AUTH_JWT) {
+            SET_SNOWFLAKE_ERROR(&conn->error, SF_STATUS_ERROR_GENERAL,
+                "authentication failed",
```

Comment:
Better to append the error message from exception.

Reply:
Updated the PR. Please take a look.